algorithms/Lesson4_CountingElements/ex_2.rb

```diff
@@ -0,0 +1,23 @@
+def solution(x, array)
+  n = array.index(x) if array.include? x
+  return -1 unless n
+
+  subarray = array[0..n].uniq
+  m = subarray.length
+  print subarray
+
+  m.times do
+    x -= 1
+    return n if x.zero?
+    if subarray.include? x
```

Comment:
and here you're using `.include` inside `times` - therefore you're making `O(n)` operation **inside** `O(n)` operation. That ofc results in `O(n*n)` performance.